### ISSUE TITLE
nodes consistently fail to be recognized as successfully created

### ISSUE THREAD

**Issue by dimhoLt** [kubernetes-incubator/kube-aws#1011]
Hello!

I'm trying to create a cluster using `kube-aws`, but I keep running into a strange issue where CloudFormation won't recognize the creation of `etcd1`.

**TL;DR** I'll provide as much information as I can, but perhaps it's not needed, so the issue is basically this: `etcd1` fails to be recognized as successful, even though the node is created and available in AWS, ~~and I think it has to do with it being created in another availability zone~~.

**Edit**: I now tested with a single private and a single public subnet, but still 3 etcd instances, and the same problem remains.

**Edit2**: Running with a single `etcd`-instance instead makes the first controller fail to create with the same errors. Apparently, the second EC2 instance definitely fails to create, regardless of type or configuration.

---

Log from  CloudFormation:
```
[...]
+00:03:01	Controlplane	CREATE_IN_PROGRESS      		Etcd0
+00:03:01	Controlplane	CREATE_IN_PROGRESS      		Etcd0                 	"Resource creation Initiated"
+00:03:05	Controlplane	CREATE_COMPLETE         		ProductionManagedPrivateSubnet2RouteToNatGateway
+00:05:26	Controlplane	CREATE_IN_PROGRESS      		Etcd0                 	"Received SUCCESS signal with UniqueId i-03fd4eafefdbb0fb3"
+00:05:27	Controlplane	CREATE_COMPLETE         		Etcd0
+00:05:31	Controlplane	CREATE_IN_PROGRESS      		Etcd1
+00:05:31	Controlplane	CREATE_IN_PROGRESS      		Etcd1                 	"Resource creation Initiated"
+00:05:10	ip-10-1-0-122.eu-west-2.compute.internal: "locksmithd.service: Current command vanished from the unit file, execution of the command list won't be resumed."
+00:07:27	ip-10-1-1-53.eu-west-2.compute.internal: "locksmithd.service: Current command vanished from the unit file, execution of the command list won't be resumed."
+00:07:52	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:08:53	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:09:55	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:10:56	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:11:58	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:12:59	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:14:01	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:15:02	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:16:04	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:17:05	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:18:07	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:19:08	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:20:10	ip-10-1-1-53.eu-west-2.compute.internal: "etcdadm-check.service: Failed with result 'exit-code'."
+00:20:33	Controlplane	UPDATE_IN_PROGRESS      		Etcd1                 	"Failed to receive 1 resource signal(s) for the current batch.  Each resource signal timeout is counted as a FAILURE."
+00:20:34	Controlplane	CREATE_FAILED           		Etcd1                 	"Received 0 SUCCESS signal(s) out of 1.  Unable to satisfy 100% MinSuccessfulInstancesPercent requirement"
+00:20:35	Controlplane	CREATE_FAILED           		production-Controlplane-PXHK8W9RK3KC	"The following resource(s) failed to create: [Etcd1]. "
```

I'm not certain how to debug this, so any help to provide better troubleshooting material is appreciated as well.

To clarify: `etcd0` is created without a problem, and when `etcd1` is about to be created, everything is successful, but it's success isn't recognized by CloudFormation / kube-aws, thus failing the `up` process.

**Things I've tried**
**Erroneously configured secgroups?**
I figured that since the issue was some sort of connectivity issue, perhaps I incorrectly set up the subnets (see bottom for cluster excerpts).

Creating a manual subnet to explicitly allow all traffic between nodes in the VPC made no difference.

**Overriding AWS limits?**
Since I'm trying to create six subnets (three private and three public; one of each in each availability zone) and the default AWS region EIP limit is 5 (even though my account is upped to 10), I tried using only two subnets. No change in upping.

**Old data in stack data / credentials?**
Removing all credentials and stacks and re-running `kube-aws render credentials` and `kube-aws render stack` and then running `up` again made no difference.

**Temporary issue in AWS network?**
I rebuilt everything, created a new S3 bucket in a new region and `up`ped there. No difference.

~~The only thing I haven't tried is simply using one subnet and one availability zone. The reason why I haven't tried it is because first of all, I really think it works (since the `etcd0` in the primary availability zone consistently creates successfully), therefore not providing any new unknown details and because I don't want to run a cluster in a single availability zone, thus not assisting in forwarding any solution.~~ Using a single private and a single public subnet in the same region gives the same error.

Thanks for reading!

Here's an excerpt of my `cluster.yaml`, showing the parts I think are applicable. If more is needed, please tell me and I'll expand.

```# This file holds all the configurations for creating the Pensionera production cluster.
#
# Template containing all configurable values values can be found here:
# https://github.com/kubernetes-incubator/kube-aws/blob/master/core/controlplane/config/templates/cluster.yaml

# Unique name of Kubernetes cluster. In order to deploy
# more than one cluster into the same AWS account, this
# name must not conflict with an existing cluster.
clusterName: production

# CoreOS release channel to use. Currently supported options: alpha, beta, stable
# See coreos.com/releases for more information
releaseChannel: stable

# CIDR for Kubernetes VPC. If vpcId is specified, must match the CIDR of existing vpc.
vpcCIDR: "10.1.0.0/16"

# Version of hyperkube image to use. This is the tag for the hyperkube image repository.
kubernetesVersion: v1.6.6_coreos.1

[...]

apiEndpoints:
- # The unique name of this API endpoint used to identify it inside CloudFormation stacks or
  # to be referenced from other parts of cluster.yaml
  name: default

  # DNS name for this endpoint, added to the kube-apiserver TLS cert
  # It must be somehow routable to the Kubernetes controller nodes
  # from worker nodes and external clients. Configure the options
  # below if you'd like kube-aws to create a Route53 record sets/hosted zones
  # for you.  Otherwise the deployer is responsible for making this name routable
  dnsName: [redacted]

  # Configuration for the load balancer serving this endpoint
  # Omit all the settings when you want kube-aws not to provision a load balancer for you
  loadBalancer:
    # All the subnets assigned to this load-balancer. Specified only when this load balancer is not reused but managed one
    # Must be omitted when `id` is specified
    subnets:
    - name: ProductionManagedPublicSubnet1
    - name: ProductionManagedPublicSubnet2
    - name: ProductionManagedPublicSubnet3

    # The Route 53 hosted zone is where the resulting Alias record is created for this endpoint
    #
    # Omitting hostedZone.id implies you must configure Route 53 hosted zone and record set or your own DNS so that worker nodes
    # are able to resolve DNS names of this API endpoint.
    #
    # Must be omitted when `id` is specified because kube-aws doesn't deal with the existing ELB by design
    hostedZone:
      # The ID of hosted zone to add the dnsName to.
      id: [redacted]

    # Set to true when you want kube-aws to create a Route53 ALIAS record set for this API load balancer for you
    # Must be omitted when `id` is specified
    createRecordSet: true

[...]

## Etcd Cluster config
## WARNING: Any changes to etcd parameters after the cluster is first created will not be applied
## during a cluster upgrade, due to concerns over data loss.
## This situation is being rectified with work towards automated management of etcd clusters
etcd:
  # Number of etcd nodes
  # (Set to an odd number >= 3 for HA control plane)
  # WARNING: You can't add etcd nodes after the first creation by modifying this and running `kube-aws update`
  # See https://github.com/kubernetes-incubator/kube-aws/issues/631 for more information
  count: 3

  # Instance type for etcd node
  instanceType: t2.medium

  # The version of etcd to be used. Set to e.g. "3.2.1" to use etcd3
  version: 3.2.9

  rootVolume:
    # Root volume size (GiB) for etcd node
    size: 30
    # Root volume type for etcd node (one of standard, io1, or gp2)
    type: gp2
    # Number of I/O operations per second (IOPS) that the etcd node's root volume supports. Leave blank if etcdRootVolumeType is not io1
    iops: 0

  dataVolume:
    # Data volume size (GiB) for etcd node
    # if etcdDataVolumeEphemeral=true, this value is ignored. The size of ephemeral volumes is not configurable.
    size: 30
    # Data volume type for etcd node (one of standard, io1, or gp2)
    type: gp2
    # Number of I/O operations per second (IOPS) that the etcd node's data volume supports. Leave blank if etcdDataVolumeType is not io1
    iops: 0
    # Encrypt the Etcd Data volume.  Set to true for encryption.  This does not work for etcdDataVolumeEphemeral.
    encrypted: true
    # Use ephemeral instance storage for etcd data volume instead of EBS?
    # (Recommended set to true for high-throughput control planes)
    # CAUTION: Currently broken. Please don't turn this on until it is fixed in https://github.com/kubernetes-incubator/kube-aws/pull/417
    ephemeral: false

  snapshot:
    # Set to true to periodically take an etcd snapshot
    # Beware that this can be enabled only for etcd 3+
    # Please carefully test if it works as you've expected when being enabled for your production clusters
    automated: true

  disasterRecovery:
    # Set to true to automatically execute a disaster-recovery process whenever etcd node(s) seemed to be broken for a while
    # Beware that this can be enabled only for etcd 3+
    # Please carefully test if it works as you've expected when being enabled for your production clusters
    automated: true

  subnets:
    # References subnets defined under the top-level `subnets` key by their names
    - name: ProductionManagedPrivateSubnet1
    - name: ProductionManagedPrivateSubnet2
    - name: ProductionManagedPrivateSubnet3

[...]

subnets:
  # Private subnet managed by kube-aws
  - name: ProductionManagedPrivateSubnet1
    private: true
    availabilityZone: eu-west-2a
    instanceCIDR: "10.1.0.0/24"

  # Private subnet managed by kube-aws
  - name: ProductionManagedPrivateSubnet2
    private: true
    availabilityZone: eu-west-2b
    instanceCIDR: "10.1.1.0/24"

  # Private subnet managed by kube-aws
  - name: ProductionManagedPrivateSubnet3
    private: true
    availabilityZone: eu-west-2c
    instanceCIDR: "10.1.2.0/24"

  # Public subnet managed by kube-aws
  - name: ProductionManagedPublicSubnet1
    private: false
    availabilityZone: eu-west-2a
    instanceCIDR: "10.1.10.0/24"

  # Public subnet managed by kube-aws
  - name: ProductionManagedPublicSubnet2
    private: false
    availabilityZone: eu-west-2b
    instanceCIDR: "10.1.11.0/24"

  # Public subnet managed by kube-aws
  - name: ProductionManagedPublicSubnet3
    private: false
    availabilityZone: eu-west-2c
    instanceCIDR: "10.1.12.0/24"
```

**Comment by mumoshu** [comment]
Hi! Thanks for trying kube-aws, and the detailed report.

Would you mind sharing outputs from `journalctl -u etcdadm` and just
`journalctl` on the failing etcd node?

2017年11月18日(土) 2:12 dimhoLt <notifications@github.com>:

**Comment by dimhoLt** [comment]
Sure! I'm trying to connect, but I keep being prompted for a password, even though (I think) I'm using the proper certificate. I guess the users I'm trying don't exist (AWS' suggested usernames `root`, `ec2-user` etc). What are the user names set on the instances?

I'm trying to run `$ ssh -i private/ca.key.pem root@ec2-1-2-3-4.eu-central-1.compute.amazonaws.com`

**Comment by mumoshu** [comment]
Thanks.

The username must be `core`!

Also, specify your private
key correspond to the keyPairName you've specified in cluster.yaml
kube-aws does't manage ssh key for you but EC2 does.
2017年11月18日(土) 10:30 dimhoLt <notifications@github.com>:

**Comment by dimhoLt** [comment]
Aah, of course. I keep forgetting that it's coreOS. Also, good thing about the SSH key - I was using the certificate for debugging purposes, good thing it's noted here that you shouldn't log in with the root cert.

When attempting to access the etcd, I get a timeout:

    $ ssh -i ~/.ssh/id_rsa core@ec2-1-2-3-4.eu-central-1.compute.amazonaws.com
    ssh: connect to host ec2-1-2-3-4.eu-central-1.compute.amazonaws.com port 22: Operation timed out

I made sure to open up port 22 for myself in the security group before I connected, and it works properly with another cluster I managed to set up before (with the same config), so I know I'm doing it properly.

Worth noting is also that I couldn't connect to `etcd0` either, with the same sec group.

**Comment by mumoshu** [comment]
Thx again!
Can you ssh into the etcd node via an another EC2 instance launched by you?

2017年11月18日(土) 11:19 dimhoLt <notifications@github.com>:

**Comment by dimhoLt** [comment]
I managed to connect through a manually started container.

`journalctl -u etcdadmin` gives this output from the failed etcd instance:

    core@ip-10-1-1-119 ~ $ journalctl -u etcdadm
    -- Logs begin at Sat 2017-11-18 14:15:33 UTC, end at Sat 2017-11-18 15:53:02 UTC. --
    -- No entries --

If I run just `journalctl` I get thousands of lines of logs, so it's difficult to post here, but I've narrowed out all the error messages I found. Please tell me if you need more or the full log file.

Here are the errors:

    core@ip-10-1-1-119 ~ $ journalctl
    [...]
    Nov 18 14:15:33 localhost kernel: Cannot get hvm parameter CONSOLE_EVTCHN (18): -22!
    [...]
    Nov 18 14:15:33 localhost kernel: Cannot get hvm parameter CONSOLE_EVTCHN (18): -22!
    [...]
    Nov 18 14:15:33 localhost systemd-udevd[231]: Invalid rule /usr/lib64/udev/rules.d/73-seat-late.rules:15: RUN{builtin}: 'uaccess' unknown
    [...]
    Nov 18 14:15:33 localhost systemd-udevd[231]: Invalid rule /usr/lib64/udev/rules.d/73-seat-late.rules:15: RUN{builtin}: 'uaccess' unknown
    [...]
    Nov 18 14:15:35 localhost ignition[394]: failed to fetch config: not a config (found coreos-cloudinit script)
    [...]
    Nov 18 14:15:35 localhost ignition[444]: failed to fetch config: not a config (found coreos-cloudinit script)
    [...]
    Nov 18 14:15:39 localhost systemd-udevd[635]: Invalid rule /usr/lib64/udev/rules.d/73-seat-late.rules:15: RUN{builtin}: 'uaccess' unknown
    [...]
    Nov 18 14:15:39 localhost kernel: piix4_smbus 0000:00:01.3: SMBus base address uninitialized - upgrade BIOS or use force_addr=0xaddr
    [...]
    Nov 18 14:15:42 localhost systemd-udevd[635]: Invalid rule /usr/lib64/udev/rules.d/73-seat-late.rules:15: RUN{builtin}: 'uaccess' unknown
    [...]
    Nov 18 14:15:44 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: Reboot strategy "best-effort" is deprecated and will be removed in the future.
    Nov 18 14:15:44 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: Please explicitly set the reboot strategy to one of [off reboot etcd-lock]
    Nov 18 14:15:44 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: See https://coreos.com/os/docs/latest/update-strategies.html for details on configuring reboot strategies.
    [...]
    Nov 18 14:15:47 ip-10-1-1-119.eu-central-1.compute.internal systemd-udevd[635]: Invalid rule /usr/lib64/udev/rules.d/73-seat-late.rules:15: RUN{builtin}: 'uaccess' unknown
    Nov 18 14:15:49 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: [etcd.service etcd2.service] are inactive
    Nov 18 14:15:49 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 10s.
    [...]
    Nov 18 14:16:00 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 20s.
    [...]
    Nov 18 14:16:20 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 40s.
    [...]
    Nov 18 14:17:00 ip-10-1-1-119.eu-central-1.compute.internal locksmithd[727]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 1m20s.
    [...]
    Nov 18 14:17:10 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: Failed to start etcd (System Application Container).
    [...]
    Nov 18 14:17:25 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: Failed to start etcd (System Application Container).
    Nov 18 14:17:25 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: Dependency failed for etcdadm update status.
    Nov 18 14:17:25 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcdadm-update-status.service: Job etcdadm-update-status.service/start failed with result 'dependency'.
    Nov 18 14:17:25 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcd-member.service: Unit entered failed state.
    Nov 18 14:17:25 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcd-member.service: Failed with result 'exit-code'.
    [...]
    Nov 18 14:17:34 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcdadm-check.service: Control process exited, code=exited status=3
    Nov 18 14:17:34 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: Failed to start etcd health check.
    Nov 18 14:17:34 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcdadm-check.service: Unit entered failed state.
    Nov 18 14:17:34 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcdadm-check.service: Failed with result 'exit-code'.

After this, the restart procedure is repeated, eventually printing these logs several times:

    Nov 18 14:17:38 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcd-member.service: Main process exited, code=exited, status=1/FAILURE
    Nov 18 14:17:38 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: Failed to start etcd (System Application Container).
    Nov 18 14:17:38 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: Dependency failed for etcdadm update status.
    Nov 18 14:17:38 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcdadm-update-status.service: Job etcdadm-update-status.service/start failed with result 'dependency'.
    Nov 18 14:17:38 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcd-member.service: Unit entered failed state.
    Nov 18 14:17:38 ip-10-1-1-119.eu-central-1.compute.internal systemd[1]: etcd-member.service: Failed with result 'exit-code'.

These logs then repeat until the container is terminated after the 15 minute timeout.

I also get a bunch of these throughout the logs, aside from the errors reported. Don't know if they're significant, but seemed related:

    Nov 18 14:17:37 ip-10-1-1-119.eu-central-1.compute.internal kernel: SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)

**Comment by mumoshu** [comment]
Thanks again!

As far as I remember, the mount invalid error was there even in a successful nodes and clusters.

Anyway, I wonder if you could find something in etcm-member's log because the service seems to be the root of all the failing service dependeocies in this case.
You can run `journalctl -u etcd-member` for that.

Also, in case it helps, you can paste lengthy logs in GitHub's Gist and share the links to them so that you won't make this thread unreadable.

**Comment by dimhoLt** [comment]
Thanks for taking your time, really appreciate it!

The output for the `etcd-member` user can be found [in this gist](https://gist.github.com/dimhoLt/7e19b1e18bb6a47710b86e605094b5cd).

**Comment by redbaron** [comment]
@dimhoLt ,

```
Nov 19 14:57:34 ip-10-1-1-127.eu-central-1.compute.internal etcd-wrapper[3233]: 2017-11-19 14:57:34.827399 I | embed: peerTLS: cert = /etc/ssl/certs/etcd.pem, key = /etc/ssl/certs/etcd-key.pem, ca = , trusted-ca = /etc/ssl/certs/etcd-trusted-ca.pem, client-cert-auth = false
Nov 19 14:57:34 ip-10-1-1-127.eu-central-1.compute.internal etcd-wrapper[3233]: 2017-11-19 14:57:34.827691 C | etcdmain: tls: failed to find any PEM data in key input
```

can you please tripple check that your pem files are all good?

**Comment by dimhoLt** [comment]
Thanks for your comment, @redbaron. I've manually checked all the credential certs, as well as the intermediate certs used to generate the credentials, and they all seem fine from what I can tell. I tried upping again, and the same error occurs.

Tomorrow, I'll try to redo everything from scratch; generating new CA certs as well and hope for the best... I really need to get this environment up and running soon.

**Comment by mumoshu** [comment]
@dimhoLt Hi, thanks again for the response.

Several comments:

- Could you verify all the pem files under `credentials/` are non-zero sized?
   - If so, it would be an issue on node-side.
- Could you verify that your KMS master key specified via `kmsKeyArn` in cluster.yaml is allowed to be called from the IAM role assigned to the etcd node?
  - You can verify it by reading `journalctl -u flanneld.service` and confirming there are no messages  indicating decryption errors.
  - You can run `docker run -v /etc/kubernetes/:/etc/kubernetes/ -it --rm coreos/awscli bash` and try to execute something like `aws kms decrypt --ciphertext-blob "$(cat /etc/kubernetes/etcd-key.pem.enc)"` (totally untested but the idea would be valid)

Basically, as long as the etcd node has correct permissions to decrypt keys and the keys are properly deployed to the etcd node from your local machine to the etcd node via the s3 bucket, it should work. The error like `etcdmain: tls: failed to find any PEM data in key input` indicates one ore more in those assumptions are not met.

**Comment by dimhoLt** [comment]
I checked the `credentials` folder again, and the `worker-ca-key.pem` is, in fact, 0 bytes in size. This is curious... it is generated by `kube-aws render credentials`, no? If so, how did I misconfigure it? All other certs are valid.

`$ journalctl -u flanneld.service` outputs this:

     core@ip-10-1-1-164 ~ $ journalctl -u flanneld.service
    -- Logs begin at Mon 2017-11-20 10:47:46 UTC, end at Mon 2017-11-20 11:05:53 UTC. --
     -- No entries --

From which server should I run the `docker`-command? The `etcd`-nodes don't have an `/etc/kubernetes` (or any other `kubernetes`) directory.

I am currently renewing all my root CA information etc to make sure I have really done everything there properly. Will get back asap with details on my success.

**Comment by mumoshu** [comment]
@dimhoLt Thanks for the response again.

You are correct - `/etc/kubernetes/` should be `/etc/ssl/certs` on etcd nodes.
Also, `journalctl -u flanneld.service` should be `journalctl -u decrypt-tls-assets` on etcd nodes.

The empty worker-ca-key.pem issue may related to #1003 which is I'm investigating right now. Perhaps just `kube-aws render`, instead of `kube-aws render stack` + `kube-aws render credentials` would preserve the worker-ca-key.pem to a non-empty state?

**Comment by dimhoLt** [comment]
Despite the `worker-ca-key.pem` still being empty, I tried to `up` with the newly generated credentials, and now it seems to work! Good call guys, many thanks for your help! I'm impressed by your dedication and assistance, @mumoshu and @redbaron, having read a bunch of other issues in this repository. Very good job!

**Solution**
To clarify for future readers of this thread: the problem was that the root certificates I used to generate the credentials were apprently erroneous, I'm not sure exactly how, since all credentials properly generated. To solve the problem, I did this:

1. I redid all of my root certificates, in accordance with [this tutorial](https://jamielinux.com/docs/openssl-certificate-authority/introduction.html) (note that I used a password-less intermediary certificates by running `openssl genrsa -out intermediate/private/intermediate-pwless.key.pem 4096 -nodes`, dropping `-aes256` and adding `-nodes`)

2. Completely removed the `credentials`, `userdata` and `stack-templates` folders to avoid having old data.

3. Re-ran `$ kube-aws render credentials` with the new `ca-chain.cert.pem`-cert and `intermediate-pwless.key.pem` from step 1.

4. Re-ran `$ kube aws up --s3-uri=s3://[my bucket]`.

**Comment by dimhoLt** [comment]
In a side note, even though the cluster is up, all proxy and dashboard pods keep crashing. Running `kubectl logs` gives me `Unknown flag: --config`. It's supposed to be called `--kubeconfig`? It's off topic, but a push in the right direction would be greatly appreciated as googling have me nothing...

**Comment by mumoshu** [comment]
@dimhoLt Thanks for your patience!
Would you mind telling me which version of kube-aws you are using?
Perhaps your k8s version `v1.6.6_coreos.1` specified in cluster.yaml is too old - guess it works if you specify e.g. v1.7.10_coreos.0?

**Comment by dimhoLt** [comment]
Yup, that worked! I've been trying to find the kubernetesVersions to set, but I wasn't able to. It works now though. I can't connect to the dashboard though (it times out), but I guess that's more of a StackOverflow issue, so I'll open it there.

Many thanks for your help and time, really appreciate it!

**Comment by mumoshu** [comment]
@dimhoLt Thanks for the confirmation!
Perhaps #1001 would help?

**Comment by dimhoLt** [comment]
Actually, I had put the controller nodes in the private network, disallowing access to the dashboard, but when I fixed that and re-`up`ped the cluster, I did get that same error hahah.

I'll take a look at that. Feel free to close this issue as re-generating my root CAs resolved it, and again - many thanks for your help and efforts.

**Comment by bennythejudge** [comment]
Hi,
I am having what looks like the same problem and I have gone through the steps suggested by @dimhoLt  but I am still having the same problem.

I get to the point where one EC2 for etcd is created as part of the ASG.
I can ssh into the EC2 and `journalctl -f` shows:

```
Apr 24 21:10:29 ip-10-0-1-25.ec2.internal etcd-wrapper[1465]: 2018-04-24 21:10:29.069632 I | raft: 1e76aedd9da2314a [logterm: 1, index: 3] sent MsgVote request to 77415e3c988d618d at term 23
Apr 24 21:10:29 ip-10-0-1-25.ec2.internal etcd-wrapper[1465]: 2018-04-24 21:10:29.094539 W | rafthttp: health check for peer 1de22eb04c509668 could not connect: dial tcp 18.205.68.14:2380: i/o timeout
Apr 24 21:10:29 ip-10-0-1-25.ec2.internal etcd-wrapper[1465]: 2018-04-24 21:10:29.095916 W | rafthttp: health check for peer 77415e3c988d618d could not connect: dial tcp 52.20.194.104:2380: i/o timeout
Apr 24 21:10:29 ip-10-0-1-25.ec2.internal etcd-wrapper[1465]: 2018-04-24 21:10:29.095948 E | etcdserver: publish error: etcdserver: request timed out
```

I don't know what it expects to find at those IP addresses, but they don't exist afaik.

I have followed the basic instructions and I have a domain, a hosted zone and a KMS resource, I have generated the cluster.yaml using a locally recompiled kube-aws due to the [issue with iops](https://github.com/kubernetes-incubator/kube-aws/issues/1170) and then generated my own CA certs etc.

My `cluster.yaml`: https://gist.github.com/bennythejudge/386c1dd5b717a7581f6cbbc9041b12eb

Any idea?

Thanks
